src/algorithms/time_evolution/fullenv_truncation.jl

```diff
@@ -0,0 +1,251 @@
+        |   ┌---┐       |   |   ┌----┐              |
+        └---|   |-→   --┘   └---|    |- u†→     ---┘
+            | B |         =     |benv|
+        ┌---|   |-←   --┐   ┌---|    |- u ←      ---┐
+        |   └---┘       |   |   └----┘              |
+        └---------------┘   └-----------------------┘
+```
+Then (each index corresponds to a pair of fused indices)
+```
+    F(r,r†) = |p† r|² / (r† B r)
+            = (r† p) (p† r) / (r† B r)
+```
+which is maximized when
+```
+    ∂F/∂r̄ * (r† B r)²
+    = p (p† r) (r† B r) - |p† r|² (B r) = 0
+```
+Note that `B` is positive (consequently `B† = B`). 
+Then the solution for the vector `r` is
+```
+    r = B⁻¹ p
+```
+We can verify that (using `B† = B`)
+```
+    ∂F/∂r̄ * (r† B r)²
+    = p (p† B⁻¹ p) (p† B⁻¹ B B⁻¹ p) - |p† B⁻¹ p|² (B B⁻¹ p) 
+    = 0
+```
+Then the bond matrix `u s v†` is updated by truncated SVD:
+```
+    - u ← r -    ==>    - u ← s ← v† -
+```
+
+## Optimization of `l`
+
+The process is entirely similar. 
+Define the vector `p` and the positive map `B` as
+```
+        ┌---------------┐   ┌-----------------------┐
+        |   ┌---┐       |   |   ┌----┐              |
+        └---|   |-------┘   └---|    |------b0† ----┘
+            | p†|         =     |benv|
+        ┌---|   |-  ←-o-┐   ┌---|    |--    ←-o- v†-┐
+        |   └---┘       |   |   └----┘              |
+        └---------------┘   └-----------------------┘
+
+        ┌---------------┐   ┌-----------------------┐
+        |   ┌---┐       |   |   ┌----┐              |
+        └---|   |-  →-o-┘   └---|    |--    →-o- v -┘
+            | B |         =     |benv|
+        ┌---|   |-  ←-o-┐   ┌---|    |--    ←-o- v†-┐
+        |   └---┘       |   |   └----┘              |
+        └---------------┘   └-----------------------┘
+```
+Here `o` is the parity tensor (twist) necessary for fermions. 
+Then (each index corresponds to a pair of fused indices)
+```
+    F(l,l†) = |p† l|² / (l† B l)
+```
+which is maximized when
+```
+    l = B⁻¹ p
+```
+Then the bond matrix `u s v†` is updated by SVD:
+```
+    - l ← v† -   ==>    - u ← s ← v† -
+```
+
+## Returns
+
+The SVD result of the new bond matrix `u`, `s`, `vh`.
+"""
+function fullenv_truncate(
+    b0::AbstractTensorMap{T,S,1,1}, benv::BondEnv{T,S}, alg::FullEnvTruncation
+) where {T<:Number,S<:ElementarySpace}
+    verbose = (alg.check_interval > 0)
+    # `benv` is assumed to be positive; here we only check codomain(benv) == domain(benv).
+    @assert codomain(benv) == domain(benv)
+    time00 = time()
+    # initialize u, s, vh with truncated SVD
+    u, s, vh = tsvd(b0; trunc=alg.trscheme)
+    # normalize `s` (bond matrices can always be normalized)
+    s /= norm(s, Inf)
+    s0 = deepcopy(s)
+    Δfid, Δs, fid, fid0 = NaN, NaN, 0.0, 0.0
+    for iter in 1:(alg.maxiter)
+        time0 = time()
+        # update `← r -  =  ← s ← v† -`
+        @tensor r[-1 -2] := s[-1 1] * vh[1 -2]
+        @tensor p[-1 -2] := conj(u[1 -1]) * benv[1 -2; 3 4] * b0[3 4]
```

Comment:
```suggestion
        @plansor p[-1 -2] := conj(u[1; -1]) * benv[1 -2; 3 4] * b0[3 4]
```
This is a small trick that works for planar diagrams, and should remove the need for the `_linearmap_twist!` afterwards I think: `@plansor` works like `*` and does not put parity matrices while contracting ket with bra.

Reply:
(with the caveat that `@plansor` is strict about the semicolon in `[a b; c]`, it needs to be in the correct place)

Reply:
I still need the fermion twists when contracting `u†`, `benv` and `b0`, and only adjust the domain/codomain of the resulting tensor. From your description I think `@plansor` is not suitable here?

Reply:
I think it should all work out actually, but I should work this out properly. It's more or less based on the following intuition:
`twist` tends to show up whenever you are combining things that have to do with inner products with `@tensor`, here the U/V are unitary (orthogonal rows/columns need an inner product), and so while computing U' * something you have to add twists. For matrices (two-leg tensors), since they have total charge 0, you can actually pull this twist through, it does not matter on which leg you put this. I think you needed the twists in the first place because you are mixing `@tensor` and `*` calls, while `@plansor` writes everything in terms of `*` which tends to be what you want with inner products.

Reply:
This being said, if the current implementation works, that's definitely also fine

Reply:
The simple drop-in replacement of `@tensor` by `@plansor` (with semicolons properly added) and removal of `_linearmap_twist!` don't work. Maybe we just keep the current version.

Reply:
Thanks for trying this out!